cardano-node/src/Cardano/Tracing/Peer.hs

```diff
@@ -0,0 +1,209 @@
+  }
+
+initialNodeKernelData :: NodeKernelData blk
+initialNodeKernelData = NodeKernelData 0 0 (SNothing)
+
+setNodeKernel :: RunNode blk
+              => IORef (NodeKernelData blk)
+              -> NodeKernel IO RemoteConnectionId LocalConnectionId blk
+              -> IO ()
+setNodeKernel nodeKernIORef nodeKern = do
+  -- This is correct for Byron and Shelley.
+  MempoolCapacityBytes mempoolCapacityBytes <- STM.atomically $ getCapacity (getMempool nodeKern)
+
+  currentLedger <- STM.atomically $ ChainDB.getCurrentLedger (getChainDB nodeKern)
+
+  let maxTxSize = nodeMaxTxSize $ ledgerState currentLedger
+      actualNodeKernelData = NodeKernelData
+        { nkdMempoolCapacity = fromIntegral $ mempoolCapacityBytes `div` maxTxSize
+        , nkdMempoolCapacityBytes = fromIntegral mempoolCapacityBytes
+        , nkdKernel = SJust (LVNodeKernel nodeKern)
+        }
+  -- We do it once, so don't need an atomic updating here.
+  writeIORef nodeKernIORef actualNodeKernelData
+
+getCurrentPeers
+  :: IORef (NodeKernelData blk)
+  -> IO [Peer blk]
+getCurrentPeers nodeKernIORef = do
+  nkd <- readIORef nodeKernIORef
+  fromSMaybe mempty <$> sequence (extractPeers . getNodeKernel <$> nkdKernel nkd)
+ where
+  tuple3pop :: (a, b, c) -> (a, b)
+  tuple3pop (a, b, _) = (a, b)
+
+  getCandidates
+    :: STM.StrictTVar IO (Map peer (STM.StrictTVar IO (Net.AnchoredFragment (Header blk))))
+    -> STM.STM IO (Map peer (Net.AnchoredFragment (Header blk)))
+  getCandidates var = STM.readTVar var >>= traverse STM.readTVar
+
+  extractPeers :: NodeKernel IO RemoteConnectionId LocalConnectionId blk
+                -> IO [Peer blk]
+  extractPeers kernel = do
+    peerStates <- fmap tuple3pop <$> (   STM.atomically
+                                       . (>>= traverse readFetchClientState)
+                                       . Net.readFetchClientsStateVars
+                                       . getFetchClientRegistry $ kernel
+                                     )
+    candidates <- STM.atomically . getCandidates . getNodeCandidates $ kernel
+
+    pure $ Map.elems . flip Map.mapMaybeWithKey candidates $
+      \cid af -> Map.lookup cid peerStates <&>
+      \(status, inflight) -> Peer cid af status inflight
+
+-- | Trace peers list, it will be forwarded to an external process
+--   (for example, to RTView service).
+tracePeers
+  :: Trace IO Text
+  -> [Peer blk]
+  -> IO ()
+tracePeers tr peers = do
+  let tr' = appendName "metrics" tr
+  let tr'' = appendName "peersFromNodeKernel" tr'
+  meta <- mkLOMeta Notice Public
+  traceNamedObject tr'' (meta, LogStructured $ toObject MaximalVerbosity peers)
+
+-- | Instances for convertin [Peer blk] to Object.
+
+instance ToObject [Peer blk] where
+  toObject MinimalVerbosity _ = emptyObject
+  toObject _ [] = emptyObject
+  toObject verb xs = mkObject
+    [ "kind"  .= String "NodeKernelPeers"
+    , "peers" .= toJSON
+      (foldl' (\acc x -> toObject verb x : acc) [] xs) ]
```

Comment:
Here, the closing sqaure bracket is at the end of the line, but 10 lines below, it is on a new line.

Consistency please. Personally I prefer the closing square bracket on a new line like below. Its easier to read.